### PR TITLE
Beepsky is now electric immune

### PR DESCRIPTION
**Pull request by ghost**
No more free baton from the station/mining outpost if you have two rods and an apc. There are still multiple ways to take out beepsky, but they require a little more effort than the meta strategy. He gets a speed boost instead.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
better yet, electric shocks should overclock him and make him angry

**Comment by py01** [comment]
sounds funny but >when you code it

**Comment by SpacePrius** [comment]
electric weakness makes sense and has important implications for antags. I would honestly like to see more justification for why this change needs to be done

**Comment by py01** [comment]
Make a grille over a apc node in a hall, it becomes electrified. Punch a beepsky and kite him into the grille, he will repeatedly run it and shock himself until he dies, dropping a stun baton. This takes about ~1-2 minutes to do at roundstart, and can easily be done without anyone noticing for a free stun baton.

**Comment by carshalash** [comment]
Very nice

**Comment by PKPenguin321** [comment]
@py01 merge this https://github.com/py01/tgstation/pull/1

**Comment by py01** [comment]
eh fuck it

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -229,3 +229,4 @@
 #define ANTI_DROP_IMPLANT_TRAIT "anti-drop-implant"
 #define HIVEMIND_ONE_MIND_TRAIT "one_mind"
 #define VR_ZONE_TRAIT "vr_zone_trait"
+#define BEEPSKY_TRAIT "beepsky_trait"
```
Comment (PKPenguin321) [inline code/__DEFINES/traits.dm]: if u merge my thing undo this change too

**Comment by py01** [comment]
I probably shouldn't have assumed you tested that

**Comment by PKPenguin321** [comment]
No worries, it's an easy fix.

File: code/modules/mob/living/simple_animal/bot/secbot.dm [hunk 2]
```diff
@@ -92,6 +92,13 @@
 	walk_to(src,0)
 	last_found = world.time
 
+/mob/living/simple_animal/bot/secbot/electrocute_act()//shocks only make him angry
```
Comment (PKPenguin321) [inline code/modules/mob/living/simple_animal/bot/secbot.dm]: Replace the last bit with
`electrocute_act(shock_damage, source, siemens_coeff = 1, safety = 0, override = 0, tesla_shock = 0, illusion = 0, stun = TRUE)`

**Comment by Akrilla** [comment]
sad

**Comment by Akrilla** [comment]
I'd say just make it so he passes over grilles

File: code/modules/mob/living/simple_animal/bot/secbot.dm (hunk 2 above)
Comment (PKPenguin321) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: Also https://github.com/tgstation/tgstation/pull/45291 just got merged and might conflict a tiny bit so if it still fails use the args from that pr

**Comment by XDTM** [comment]
Make him phase through walls

Nothing stops the law

**Comment by 81Denton** [comment]
@SpacePrius Beepsky is a free stun baton since he'll straight up path into electrified grilles that can be built anywhere

File: code/modules/mob/living/simple_animal/bot/secbot.dm (hunk 2 above)
Comment (nemvar) [reply code/modules/mob/living/simple_animal/bot/secbot.dm]: Give beepsky the shock_immune trait and call parent. That's why I refactored electrocute_act in the first place, so stuff sends the god damn signal.

**Comment by Yttriums** [comment]
imsxz just cannot quit telling on themself with the YT vids

**Comment by Ty-the-Smonk** [comment]
Bluespace beepsky that teleports to arrested people WHEN

**Comment by XDTM** [comment]
It's already on my todo list
